src/libsyntax/ast.rs

```diff
@@ -770,7 +850,7 @@ impl BinOpKind {
         use BinOpKind::*;
         match *self {
             Eq | Lt | Le | Ne | Gt | Ge => true,
-            And | Or | Add | Sub | Mul | Div | Rem | BitXor | BitAnd | BitOr | Shl | Shr => false,
+            _ => false,
```

Comment:
Please keep this as is; we want compilation to fail if another variant is added.

Reply:
Sure! I've also added a comment for future wannabe-optimizers like me to keep it as is :)